src/components/molecules/Darkmode/DarkmodeContainer.jsx

```diff
@@ -0,0 +1,47 @@
+import storage from '../../../lib/storage'
+import Darkmode from './Darkmode'
+import { compose, withStateHandlers, withPropsOnChange, withHandlers, lifecycle, pure} from 'recompose';
+
+export default compose(
+  pure,
+  withStateHandlers(
+    {
+      isActive: false,
+      isSupported: null
+    },
+    {
+      toggle: ({isActive}) => (value) => ({
+        isActive: value
+      })
+    }
+  ),
+  withHandlers({
+    isSupported: () => (property = 'filter', value = 'invert(100%)') => {
+      let prop = property + ':',
+      el = document.createElement('test'),
+      mStyle = el.style
+      el.style.cssText = prop + value
+      return mStyle[property]
+    }
+  }),
+  lifecycle({
+    componentDidMount() {
+      this.props.toggle(storage.get('DarkMode') || false)
+    },
+    componentDidUpdate() {
+      storage.set('DarkMode', this.props.isActive)
+    }
+  }),
+  withPropsOnChange([ 'isSupported' ], ({ isSupported }) => {
+    if(isSupported) {
+      return {
+      cssStyles: `
```

Comment:
@Joe-Pena since we are already using the `withPropsOnChange`, we don't need the `componentDidUpdate`. `withPropsOnChange` will listen for a change on any given prop. As mentioned above, it doesn't look like `isSupported` is a prop that is being updated. We could add a listener to `isActive` though. Every time `isActive` changes, we set DarkMode and pass through the `cssStyles` prop. 

```
withPropsOnChange([ 'isActive' ], ({ isActive }) => {
    if(isActive) {
      storage.set('DarkMode', isActive)
      return {
        cssStyles: 
            ....
       }
```
No need for the else statement.

Reply:
Neat! Implemented the change and got rid of the lifecycle method.